hadoop-hdds/docs/content/feature/S3-Multi-Tenancy.md

```diff
@@ -0,0 +1,94 @@
+
+> This multi-tenant feature allows Ozone resources to be compartmentalized in different isolation zones (tenants).
+
+> This multi-tenant support will also allow users to access Ozone volumes over AWS S3 APIs (without any modifications to the APIs).
+
+## Basics
+
+1. Initial tenant creation has to be done by an Ozone cluster admin under the CLI.
+2. The Ozone cluster admin will have to assign the first user of a tenant. Once assigned, an _Access ID & Secret Key pair_ (key pair) will be generated for that user for access via S3 Gateway. 
+   - The key pair serves to authenticate the end user to the Ozone Manager (via client requests from S3 Gateway). Tenant volume is selected based on the Access ID.
+   - After successful authentication, Ozone Manager uses the underlying user name (not the Access ID) to identify the user. The user name is used to perform authorization checks in Apache Ranger.
+   - A key pair is tied to the user name it is assigned to. If a user is assigned key pairs in multiple tenants, all key pairs point to the same user name internally in Ozone Manager.
+   - A user can be only assigned one key pair in a same tenant. Ozone Manager rejects the tenant user assign request if a user is already assigned to the same tenant (i.e. when the user has already been assigned an Access ID in this tenant).
+   - Key pairs assigned to a user for different tenants are used to access buckets under different tenants. For instance, `testuser` uses `tenantone$testuser` to access `tenantone` buckets, and uses `tenanttwo$testuser` to access `tenanttwo` buckets via S3 Gateway.
+     - A bucket link can be set up if cross-tenant (cross-volume) access is desired.
+3. The Ozone cluster admin can then assign tenant admin roles to that user.
+4. Tenant admin are able to assign new users to the tenant.
+   - They can even assign new tenant admins in their tenant, if they are delegated tenant admins, which is the default. See the usage below for more details.
+   - Note that tenant admins still need to use Ozone tenant CLI to assign new users to the tenant.
+     - Once tenant admins get the Kerberos TGT (via `kinit`), they can run `user assign` command to assign new users. Ozone Manager will recognize that they are the tenant admins and allow the user to do so in their tenants.
+5. After that, users can use any S3-compatible client (awscli, Python boto3 library, etc.) to access the buckets in the tenant volume via S3 Gateway using the generated key pairs.
+
+
+## Access Control
+
+Ozone multi-tenancy relies on [Apache Ranger]({{< ref "security/SecurityWithRanger.md" >}}) to enforce access control to resources.
+
+By default, a group of Ranger policies are created when a tenant is created on the tenant volume:
+
+1. All users are able to create new buckets;
+2. Only the bucket owner (i.e. the user that creates the bucket) and tenant admins can access the bucket content. 
```

Comment:
What if the bucket owner or admin wants to allow more users to access the bucket? We discussed this offline with @prashantpogde and it would need to be handled via Ranger, but we need to make sure the background sync doesn't overwrite the change. Once we have a solution it should be documented here.